client/landing/gutenboarding/components/mshots-image/index.tsx

```diff
@@ -14,54 +14,74 @@ interface MShotsImageProps {
 	url: string;
 	alt: string;
 	'aria-labelledby': string;
+	options: MShotsOptions;
 }
 
-const mShotsParams = {
-	// viewport size (how much of the source page to capture)
-	vpw: 1200,
-	vph: 3072,
-	// size of the resulting image
-	w: 700,
-	h: 1800,
+export type MShotsOptions = {
+	vpw: number;
+	vph: number;
+	w: number;
+	h: number;
 };
 
-export function mshotsUrl( url: string, count = 0 ): string {
+export function mshotsUrl( url: string, options: MShotsOptions, count = 0 ): string {
 	const mshotsUrl = 'https://s0.wp.com/mshots/v1/';
 	const mshotsRequest = addQueryArgs( mshotsUrl + encodeURIComponent( url ), {
-		...mShotsParams,
-		// this doesn't seem to work:
-		// requeue: true, // Uncomment this line to force the screenshots to be regenerated
+		...options,
 		count,
 	} );
 	return mshotsRequest;
 }
 
+const MAXTRIES = 10;
+
 const MShotsImage = ( {
 	url,
 	'aria-labelledby': labelledby,
 	alt,
+	options,
 }: MShotsImageProps ): JSX.Element => {
-	const [ count, setCount ] = React.useState( 0 );
+	const [ count, setCount ] = React.useState( 1 );
 	const [ visible, setVisible ] = useState( false );
+	const [ opacity, setOpacity ] = useState( 0 );
+
+	const src = mshotsUrl( url, options, count );
+
+	// Hide the images while they're loading if src changes (e.g. when locale is switched)
+	useLayoutEffect( () => {
+		// Opacity is used for fade in on load
+		// Visible is used to hide the image quickly when swapping languages
+		setVisible( false );
+		setOpacity( 0 );
+	}, [ src, setVisible, setOpacity ] );
+
 	return (
 		<div className="mshots-image__container">
 			{ ! visible && <div className="mshots-image__loader"></div> }
 			<img
-				className={ classnames( 'mshots-image' ) }
-				style={ { opacity: visible ? 1 : 0 } }
+				className={ classnames(
+					'mshots-image',
+					visible ? 'mshots-image-visible' : 'mshots-image-hidden'
+				) }
+				style={ { opacity: opacity } }
 				alt={ alt }
 				aria-labelledby={ labelledby }
-				src={ mshotsUrl( url, count ) }
+				src={ src }
 				onLoad={ ( e ) => {
-					// Test against mshots h value
-					if ( e.currentTarget.naturalHeight !== mShotsParams.h ) {
-						// Only refresh 10 times.
-						if ( count < 10 ) {
-							// Triggers a target.src change
-							setTimeout( () => setCount( count + 1 ), 1000 );
+					// Test mshots h value matches the desired image
+					// The default image (https://s0.wp.com/mshots/v1/default) is around 400x300 px h
+					// but sometimes slightly off (e.g. h: 299.99)
+					if ( e.currentTarget.naturalHeight !== options.h ) {
```

Comment:
I didn't catch this in the last PR, but clever way of detecting the loading image!